Branch-SDK/src/io/branch/referral/indexing/RegisterViewBuilder.java

```diff
@@ -0,0 +1,176 @@
+package io.branch.referral.indexing;
+
+import android.app.Activity;
+
+import java.util.HashMap;
+
+import io.branch.referral.Branch;
+import io.branch.referral.BranchError;
+
+/**
+ * <p>
+ * Builder for creating a Registering content views to Branch.
+ * Builds a RegisterView request wih given options and update Branch with the request.
+ * </p>
+ */
+public class RegisterViewBuilder {
```

Comment:
Given my comment above, it would be good to abstract this into a 'ContentBuilder' that can be subclassed to RegisterView and GetShortUrl builder.